File: BackEnd/src/main/java/com/tienda_ropa/ecommerce/model/Producto.java
package com.tienda_ropa.ecommerce.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;
import java.util.*;

@Entity
@Table(name = "producto")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Producto extends Master {

    private String nombre;
    private String descripcion;

    @OneToMany(mappedBy = "producto", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference
    private Set<ImagenProducto> imagenes = new HashSet<>();

    @ManyToMany
    @JoinTable(
            name = "categoria_producto",
            joinColumns = @JoinColumn(name = "id_producto"),
            inverseJoinColumns = @JoinColumn(name = "id_categoria")
    )
    private Set<Categoria> categorias = new HashSet<>();

    @OneToMany(mappedBy = "producto", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference("producto-historico-venta")
    private Set<HistoricoPrecioVenta> historicoPreciosVenta = new HashSet<>();

    @OneToMany(mappedBy = "producto", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference("producto-historico-compra")
    private Set<HistoricoPrecioCompra> historicoPreciosCompra = new HashSet<>();


    //Esto es para realizar un pedido
    @Transient
    private Color color;

    @Transient
    private Talle talle;

}

File: FrontEnd/src/models/Producto.ts
import type Categoria from "./Categoria";
import type Color from "./Color";
import type HistoricoPrecioVenta from "./HistoricoPrecioVenta";
import type ImagenProducto from "./ImagenProducto";
import type Talle from "./Talle";

export default class Producto {
    id?: number;
    nombre: string = "";
    descripcion: string = "";
    imagenes!: ImagenProducto[];
    talle?: Talle;
    color?: Color;  
    precio?: number;
    historicoPreciosVenta!: HistoricoPrecioVenta[];
    categorias!: Categoria[];
    eliminado!: boolean;
}